homeassistant/components/ambient_network/config_flow.py

```diff
@@ -0,0 +1,169 @@
+"""Config flow for the Ambient Weather Network integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from aioambient import OpenAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LOCATION,
+    CONF_LONGITUDE,
+    CONF_MAC,
+    CONF_RADIUS,
+    UnitOfLength,
+)
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    LocationSelector,
+    LocationSelectorConfig,
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+)
+from homeassistant.util.unit_conversion import DistanceConverter
+
+from .const import (
+    API_LAST_DATA,
+    API_STATION_COORDS,
+    API_STATION_INDOOR,
+    API_STATION_INFO,
+    API_STATION_LOCATION,
+    API_STATION_MAC_ADDRESS,
+    API_STATION_NAME,
+    API_STATION_TYPE,
+    DOMAIN,
+)
+
+CONF_USER = "user"
+CONF_STATION = "station"
+
+# Roughly half a mile
+CONF_RADIUS_DEFAULT = 800
+
+
+def get_station_name(station: dict[str, Any]) -> str:
+    """Pick a station name.
+
+    Station names can be empty, in which case we construct the name from
+    the location and device type.
+    """
+    if name := station.get(API_STATION_INFO, {}).get(API_STATION_NAME):
+        return str(name)
+    location = (
+        station.get(API_STATION_INFO, {})
+        .get(API_STATION_COORDS, {})
+        .get(API_STATION_LOCATION)
+    )
+    station_type = station.get(API_LAST_DATA, {}).get(API_STATION_TYPE)
+    return f"{location}{'' if location is None or station_type is None else ' '}{station_type}"
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for the Ambient Weather Network integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Construct the config flow."""
+
+        self._longitude = 0.0
+        self._latitude = 0.0
+        self._radius = 0.0
```

Comment:
Why not store the `location` instead of these independently?
```suggestion
        self._longitude = None
        self._latitude = None
        self._radius = None
```

Reply:
I'm not sure I understand what you mean by `location`. Do you mean using `LocationInfo`? Or do you mean we should introduce a new `Location` type instead of using the three fields separately?

Reply:
I think he means a

`self.location: dict[str, Any] = {}`

Reply:
How does that make the code any more readable or better?